src/app/organizations/settings/account.component.ts

```diff
@@ -71,6 +73,13 @@ export class AccountComponent {
         } catch { }
     }
 
+    async submitTaxInfo() {
+        this.formPromise = this.taxInfo.submitTaxInfo();
```

Comment:
You need a new promise variable. Don't share it with another form. I think this would activate its spinners and such. Create `taxFormPromise` or something.